docs/Bitbucket.md

```diff
@@ -0,0 +1,107 @@
+| `pipelines.default.step.image.username` | ✅ | The username of the user for private Docker images hosted on a given registry. Translated to the `username` property within the [docker-login-buildkite-plugin](https://github.com/buildkite-plugins/docker-login-buildkite-plugin) |
+| `pipelines.default.step.image.password` | ✅ | The password of the user for private Docker images hosted on a given registry. Translated to the `password` property within the [docker-login-buildkite-plugin](https://github.com/buildkite-plugins/docker-login-buildkite-plugin). Be sure to utilise [best practices](https://buildkite.com/docs/pipelines/secrets) for storing and accessing secrets within a Buildkite pipeline! |
+| `pipelines.default.step.max-time` | ✅ | The maximum allowable time that a step within a Bitbucket pipeline is able to run for. Translates to the corresponding Buildkite pipelines' command step `timeout_in_minutes` attribute. |
+| `pipelines.default.step.name` | ✅ | The name of a specific step within a Bitbucket pipeline. Translates to a Buildkite command step's `label`. |
+| `pipelines.default.step.oidc` | ✅ | Open ID Connect configuration that will be applied for this Bitbucket pipeline step. The generated command step in the corresponding Buildkite pipeline will [request](https://buildkite.com/docs/agent/v3/cli-oidc#request-oidc-token) an OIDC token and export it into the job environment as `BITBUCKET_STEP_OIDC_TOKEN` (to be passed to `sts` to assume an AWS role for example) |
+| `pipelines.default.step.runs-on` | ✅ | Allocating the Bitbucket pipeline to run on a self-hosted runner with the specific label. All `runs-on` values will be set as agent [tags](https://buildkite.com/docs/pipelines/defining-steps#targeting-specific-agents) in the Buildkite command step for targeting on specific Buildkite agents within an organization. |
+| `pipelines.default.step.services` | ✅ | The name of one or more services defined at `definitions.services.<name>` that will be applied for this step. Translated to utilise the service configuration with the [docker-compose-buildkite-plugin](https://github.com/buildkite-plugins/docker-compose-buildkite-plugin). |
+| `pipelines.default.step.script` | ✅ | The individual commands that make up a specific step. Each is translated into a singular command within the `commands` key of a Buildkite command step. |
+| `pipelines.default.step.script.pipe` | ❌ | Re-usable modules to make configuration in Bitbucket pipelines easier and modular. Consider using [Buildkite Plugins](https://buildkite.com/docs/plugins) and/or [Pipeline Templates](https://buildkite.com/docs/pipelines/templates) in configuring re-usable configuration accross a Buildkite organization. |
```

Comment:
instead of templates I would suggest reviewing existing plugins for corresponding functionality

Reply:
Linked instead now to the plugins directory 👍